book/src/dev/rfcs/drafts/0005-treestate.md

```diff
@@ -179,9 +179,18 @@ time).
 
 IMPORTANT: we need to save the incremental merkle tree / serialized nodes for:
 
-Sapling tip block
-~all Sprout blocks
-chains we're tracking in memory within the reorg limit, for both
+Orchard tip: 
+- When finalizing state, for Orchard, the tip of the chain (latest finalized block) saves a serialization of the Orchard Note Commitment Tree
+- When non-finalized state is being updated, each non-finalized chain gets its own copy of the Orchard Note Commitment Tree, cloned from the finalized block note commitment tree of its parent block, and then subsequently updated/extended as new Orchard note commitments are discovered when processing blocks.
```

Comment:
Let's use the terminology and operations from the non-finalized state RFC:
https://github.com/ZcashFoundation/zebra/blob/treestate-rfc-updates/book/src/dev/rfcs/0005-state-updates.md#fn-commit_blockmut-self-block-arcblock

During a fork, we can either rollback (if that's possible), or delete and re-calculate from the finalized tip. We don't need to worry about the extra work of a re-calculation, because:
- it's at most ~100 blocks
- chain forks are rare
- on Mainnet, chain forks are expensive, due to proof of work, so it's hard to use them in an attack

(And we can create a cache or per-block copy if we need to, as a follow-up task.)

```suggestion
- Each non-finalized chain gets its own copy of the Orchard note commitment tree, cloned from the note commitment tree of the finalized tip or fork root.
- When a block is added to a non-finalized chain tip, the Orchard note commitment tree is updated with the note commitments from that block.
- When a block is rolled back from a non-finalized chain tip... (TODO)
```